Makefile

```diff
@@ -0,0 +1,36 @@
+# -- linting --------------------------------------------------------------------------------------
+
+.PHONY: clippy
+clippy: ## Runs Clippy with configs
+	cargo clippy -- -D warnings
+
+.PHONY: fix
+fix: ## Runs Fix with configs
+	cargo +nightly fix --allow-staged --allow-dirty --all-targets
+
+.PHONY: format
+format: ## Runs Format using nightly toolchain
+	cargo +nightly fmt --all
+
+.PHONY: format-check
+format-check: ## Runs Format using nightly toolchain but only in check mode
+	cargo +nightly fmt --all --check
+
+.PHONY: lint
+lint: format fix clippy ## Runs all linting tasks at once (Clippy, fixing, formatting)
```

Comment:
I'd be a bit surprised if running `make lint` made changes to the source tree (which is what the `fix` task will do if it finds something to fix), I'd suggest we omit that task here.